plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,221 @@
+        type: str
+  scheduling:
+    description:
+      - Schedule the job to start at a later time
+    type: dict
+    suboptions:
+      start_at:
+        description:
+          - Schedule the job for a future time
+        type: str
+      start_before:
+        description:
+          - Indicates that the action should be cancelled if it cannot be started before this time.
+        type: str
+  concurrency_control:
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template: "Run Command - SSH Default"
+    ssh:
+      effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark: 'active'
+    command: 'pwd'
+    job_template: "Run Command - Ansible Default"
+    recurrence:
+      cron_line: "30 2 * * *"
+    concurrency_control:
+      concurrency_level: 2
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanTaxonomicEntityAnsibleModule,
+)
+
+ssh_foreman_spec = {
+    'effective_user': dict(),
+}
+
+recurrence_foreman_spec = {
+    'cron_line': dict(),
+    'max_iteration': dict(type='int'),
+    'end_time': dict(),
+}
+
+scheduling_foreman_spec = {
+    'start_at': dict(),
+    'start_before': dict(),
+}
+
+concurrency_control_foreman_spec = {
+    'time_span': dict(type='int'),
+    'concurrency_level': dict(type='int'),
+}
+
+
+class ForemanJobInvocationModule(ForemanTaxonomicEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanJobInvocationModule(
+        foreman_spec=dict(
+            search_query=dict(),
+            bookmark=dict(type='entity'),
+            job_template=dict(required=True, type='entity'),
+            targeting_type=dict(default='static_query', choices=['static_query', 'dynamic_query']),
+            randomized_ordering=dict(type='bool'),
+            feature=dict(),
+            command=dict(),
+            inputs=dict(type='dict'),
+            execution_timeout_interval=dict(type='int'),
+            ssh=dict(type='dict', foreman_spec=ssh_foreman_spec),
+            recurrence=dict(type='dict', foreman_spec=recurrence_foreman_spec),
+            scheduling=dict(type='dict', foreman_spec=scheduling_foreman_spec),
+            concurrency_control=dict(type='dict', foreman_spec=concurrency_control_foreman_spec),
```

Comment:
some of these were `type=dict`, some `type=nested_list.` IMHO (by looking at https://github.com/theforeman/foreman_remote_execution/blob/master/app/controllers/api/v2/job_invocations_controller.rb#L40-L60) they all should be `type=dict`, but that actually does not support `foreman_spec`. we could use `options=…` instead, or teach the f_spec helper to understand `type=dict`?

@mdellweg what do you think?

Reply:
(fwiw, the tests include `concurrency_control`, but the recorded api tape does not)

Reply:
do you mean `nested_dict`? interesting idea.

Reply:
It's not nested as such. it's just a dict that needs to be passed to the API.

Reply:
@evgeni @mdellweg  so I changed from `foreman_spec=` to `options=` and re-recorded the tapes. Is there anything more I should do to get this in?

Reply:
I'll have a look on Monday